### ISSUE TITLE
✨ Add a way to start the bot from asynchronous function

### ISSUE THREAD

**Issue by beastmatser** [Pincer-org/Pincer#152]
**Is your feature request related to a problem? Please describe.**
When I was editing the mCoding bot to only use a global aiohttp session, I stumbled upon an error when trying to run this code:
```py
async def main():
    async with aiohttp.ClientSession() as session:
        bot.run()


asyncio.run(main())
```
Error: `RuntimeError: This event loop is already running`

**Describe the solution you'd like**
An (asynchronous) method that allows this behaviour.
Like this:
```py
async def main():
    async with aiohttp.ClientSession() as session:
        await bot.start()


asyncio.run(main())
```

**Describe alternatives you've considered**
Use a method that always checks if the session exists, if not create one, else return the existing one.

**Comment by Arthurdw** [comment]
Because of the way how asyncio works, I don't think we are able to supply your needs. But if you just want to pass an existing asyncio loop. You can start the bot with `start_loop`. 
_(https://pincer.readthedocs.io/en/latest/pincer.core.html#pincer.core.gateway.Dispatcher.start_loop)_

But, the limitation can be bypassed very easy by just creating a separate Thread.

I would keep your sessions from the bot separate with other things, as the other session can be blocking the bot which would result in unexpected behavior.

So in short, don't share the bot session with other services/tasks and if you need to have two _(or more)_ sessions just create a seperate thread for them!
